lib/Parser/OglParser.php

```diff
@@ -23,32 +23,34 @@
 class OglParser
 {
     private $hints;
+    private $matrix;
 
-    public function __construct($hints) {
+    public function __construct($hints, $matrix) {
         $this->hints = $hints;
+        $this->matrix = $matrix;
     }
 
-    public function parse($filename) {
+    public function parse($filename, $commit = null) {
         $handle = fopen($filename, "r");
```

Comment:
OK, let's try it your way (after all you're the one with more experience in PHP ;)). But allow me to told you "I told you so" if a problem arise ;)